File: debug-mcp-vscode/src/ui/ConfirmationDialog.ts
import * as vscode from "vscode";
import { BridgeClient } from "../bridge/BridgeClient";
import { BridgeMessage, MessageTypes } from "../bridge/MessageProtocol";

export async function showBreakpointConfirmation(bridge: BridgeClient, message: BridgeMessage) {
  const choice = await vscode.window.showWarningMessage(
    "AI Debugger hit a breakpoint and wants to inspect variables.",
    { modal: true },
    "View Variables",
    "Let AI Analyze",
    "Continue",
    "Step Over",
    "Stop Debug"
  );
  if (!choice) {
    bridge.send({
      type: MessageTypes.UserRejectContinue,
      confirmationId: message.confirmationId,
      sessionId: message.sessionId
    });
    return;
  }
  if (choice === "Continue") {
    bridge.send({
      type: MessageTypes.UserConfirmContinue,
      confirmationId: message.confirmationId,
      sessionId: message.sessionId,
      action: "continue"
    });
    await vscode.commands.executeCommand("workbench.action.debug.continue");
  }
  if (choice === "Step Over") {
    bridge.send({
      type: MessageTypes.UserConfirmContinue,
      confirmationId: message.confirmationId,
      sessionId: message.sessionId,
      action: "step_over"
    });
    await vscode.commands.executeCommand("workbench.action.debug.stepOver");
  }
  if (choice === "Stop Debug") {
    await vscode.commands.executeCommand("workbench.action.debug.stop");
  }
}

File: debug-mcp-idea/src/main/kotlin/ui/ConfirmationDialog.kt
package ui

import bridge.BridgeClient
import bridge.BridgeMessage
import bridge.MessageTypes
import com.intellij.openapi.ui.Messages

fun showBreakpointConfirmation(bridge: BridgeClient, message: BridgeMessage) {
    val result = Messages.showDialog(
        "AI Debugger hit a breakpoint and wants to inspect variables.",
        "AI Debugger",
        arrayOf("View Variables", "Let AI Analyze", "Continue", "Step Over", "Stop Debug"),
        2,
        Messages.getQuestionIcon()
    )
    if (result == 2 || result == 3) {
        bridge.send(
            BridgeMessage(
                type = MessageTypes.UserConfirmContinue,
                sessionId = message.sessionId,
                payload = mapOf(
                    "confirmationId" to message.payload["confirmationId"],
                    "action" to if (result == 2) "continue" else "step_over"
                )
            )
        )
    } else if (result == -1) {
        bridge.send(
            BridgeMessage(
                type = MessageTypes.UserRejectContinue,
                sessionId = message.sessionId,
                payload = mapOf("confirmationId" to message.payload["confirmationId"])
            )
        )
    }
}
